pkg/controller/masteruserrecord/sync.go

```diff
@@ -220,6 +222,27 @@ func (s *Synchronizer) alignReadiness() bool {
 			return false
 		}
 	}
+
+	s.record.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(s.record.Status.Conditions, toBeProvisioned())
+	if condition.IsNotTrue(s.record.Status.Conditions, toolchainv1alpha1.MasterUserRecordUserProvisionedNotificationCreated) {
+		notification := &toolchainv1alpha1.Notification{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      s.record.Name + "-provisioned",
+				Namespace: s.record.Namespace,
+			},
+			Spec: toolchainv1alpha1.NotificationSpec{
+				UserID:   s.record.Spec.UserID,
+				Template: "userprovisioned",
```

Comment:
@alexeykazakov sounds good, template loading gets this name from the directory name. i think keeping it as a const is a good idea - where should we store this const? it would need to be used by the controller as well in order to call `GetNotificationTemplate` in `notification_generator.go`

Reply:
It's not only about keeping that template name as a const. It's also about making sure we are referring to an existing template here. So I would keep the "userprovisioned" template object available in notification_generator.go (ideally imitable). So you can access it via `notificationtemplates.UserProvisioned` in the notification controller. And as `notificationtemplates.UserProvisioned.Name` here, in the MUR sync.

Or maybe `notificationtemplates.UserProvisioned()` and `notificationtemplates.UserProvisioned().Name` if we want to get access to a deep copy of the "const" template to make sure we don't accidentally modify the template in the controllers.

@sbryzak WDYT?

Reply:
+1 to Alexey's idea of making `UserProvisioned` a var, something like:

`var UserProvisioned, _, _ = GetNotificationTemplate("userprovisioned")`

On a side note, looking at the `GetNotificationTemplate()` function - I recommend we move this line:

`assets := assets.NewAssets(AssetNames, Asset)`

into the `loadTemplates()` function, below the cache check. So that you have is more like this:

```
func loadTemplates() (map[string]NotificationTemplate, error) {
	if notificationTemplates != nil {
		return notificationTemplates, nil
	}

	assets := assets.NewAssets(AssetNames, Asset)
```

This will avoid the `NewAssets()` call every time we call `GetNotificationTemplate()`

Reply:
@sbryzak, the reason i didn't do this was because it will disallow me to have 100% code coverage. By having loadTemplates(...assets...) i'm able to pass in custom assets into the func forcing errors

Reply:
@sbryzak @alexeykazakov is this what you had in mind then: d673eb3729c09fd32c2963da641ecb11a1b42a42

Reply:
You can do something like this to keep the code testable:
```
func loadTemplates() (map[string]NotificationTemplate, error) {
	if notificationTemplates != nil {
		return notificationTemplates, nil
	}
	return templatesForAssets(assets.NewAssets(AssetNames, Asset)
}

func templatesForAssets(assets assets.Assets) (map[string]NotificationTemplate, error {
	paths := assets.Names()
	notificationTemplates = make(map[string]NotificationTemplate)
	for _, path := range paths {
	    ...
```

Now you can test `templatesForAssets()` with whatever assets you want.

Reply:
sounds good, this is unrelated to this PR though - will create a defect maybe and fix it under that.

Reply:
It's a tiny change which could be easily addressed here IMO instead of going trough creating JIRA/planning/PRs, etc.

Reply:
addressed here: cd81b3da5005ec2e509e4e40d74ddc4615312a8b